packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -0,0 +1,656 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Method,
+  h,
+  Host,
+  Prop,
+  State,
+  VNode,
+  Watch,
+} from "@stencil/core";
+import {
+  focusElementInGroup,
+  getElementDir,
+  slotChangeGetAssignedElements,
+  toAriaBoolean,
+} from "../../utils/dom";
+import { connectLocalized, disconnectLocalized, LocalizedComponent } from "../../utils/locale";
+import { guid } from "../../utils/guid";
+import {
+  connectInteractive,
+  updateHostInteraction,
+  disconnectInteractive,
+  InteractiveComponent,
+  InteractiveContainer,
+} from "../../utils/interactive";
+import {
+  LoadableComponent,
+  componentFocusable,
+  setComponentLoaded,
+  setUpLoadableComponent,
+} from "../../utils/loadable";
+import {
+  connectMessages,
+  disconnectMessages,
+  setUpMessages,
+  T9nComponent,
+  updateMessages,
+} from "../../utils/t9n";
+import { getRoundRobinIndex } from "../../utils/array";
+import { CSS, DURATION, ICONS } from "./resources";
+import { CarouselMessages } from "./assets/carousel/t9n";
+import { ArrowType, AutoplayType } from "./interfaces";
+
+/**
+ * @slot - A slot for adding `calcite-carousel-item`s.
+ */
+@Component({
+  tag: "calcite-carousel",
+  styleUrl: "carousel.scss",
+  shadow: true,
+  assetsDirs: ["assets"],
+})
+export class Carousel
+  implements InteractiveComponent, LoadableComponent, LocalizedComponent, T9nComponent
+{
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * When `true`, the carousel will autoplay and controls will be displayed. When "paused" at time of initialization, the carousel will not autoplay, but controls will be displayed.
+   */
+  @Prop({ reflect: true }) autoplay: AutoplayType = false;
+
+  @Watch("autoplay")
+  autoplayWatcher(autoplay: boolean): void {
+    if (!autoplay) {
+      this.handlePause();
+    }
+  }
+
+  /**
+   * Specifies how and if the "previous" and "next" arrows are displayed.
+   */
+  @Prop({ reflect: true }) arrowType: ArrowType = "inline";
+
+  /**
+   * Specifies if the component's controls are positioned absolutely on top of slotted Carousel Items.
+   */
+  @Prop({ reflect: true }) controlOverlay = false;
+
+  /**
+   * When `true`, interaction is prevented and the component is displayed with lower opacity.
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   * Accessible name for the component.
+   */
+  @Prop() label!: string;
+
+  /**
+   *  When `autoplay` is `true`, specifies in milliseconds the length of time to display each Carousel Item.
+   */
+  @Prop({ reflect: true }) autoplayDuration = DURATION;
```

Comment:
This prop is not following alphabetical order. 

This can be done in a follow-up PR.